pkg/kubelet/server/stats/handler.go

```diff
@@ -113,31 +113,21 @@ type handler struct {
 }
 
 // CreateHandlers creates the REST handlers for the stats.
-func CreateHandlers(rootPath string, provider Provider, summaryProvider SummaryProvider, enableCAdvisorJSONEndpoints bool) *restful.WebService {
+func CreateHandlers(rootPath string, provider Provider, summaryProvider SummaryProvider) *restful.WebService {
 	h := &handler{provider, summaryProvider}
 
 	ws := &restful.WebService{}
 	ws.Path(rootPath).
 		Produces(restful.MIME_JSON)
 
-	type endpoint struct {
+	endpoints := []struct {
 		path    string
 		handler restful.RouteFunction
-	}
-
-	endpoints := []endpoint{
+	}{
+		{"", h.handleStats},
 		{"/summary", h.handleSummary},
 	}
 
-	if enableCAdvisorJSONEndpoints {
-		endpoints = append(endpoints,
-			endpoint{"", h.handleStats},
-			endpoint{"/container", h.handleSystemContainer},
-			endpoint{"/{podName}/{containerName}", h.handlePodContainer},
```

Comment:
this is mentioned in pkg/kubelet/server/auth_test.go Perhaps clean it up as well

Reply:
Aye, thanks, did that. There are some test failures that I need a little help with though :)

Reply:
Test failure: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/95763/pull-kubernetes-bazel-test/1319488038476189696

Failure:

```
--- FAIL: TestContainerInfo (0.02s)
    server_test.go:389: received invalid json data: json: cannot unmarshal number into Go value of type v1.ContainerInfo
I1023 04:30:17.123268  228861 server.go:398] Adding debug handlers to kubelet server.
```

So the test `func TestContainerInfo(t *testing.T) {` validates that this endpoint (`/{podName}/{containerName}`) returns json. Endpoint constructed here:

```
resp, err := http.Get(fw.testHTTPServer.URL + fmt.Sprintf("/stats/%v/%v", podID, expectedContainerName))
```

 It fails with `json: cannot unmarshal number into Go value of type v1.ContainerInfo`. So I'd guess that when endpoint is missing, `404` is returned and this response cannot be parsed as a json. So ultimately just remove the test.

I'm not sure if the removal of these endpoint is what expected, let me take a look at the original issue